apps/src/templates/sectionProgress/summary/VirtualizedSummaryView.jsx

```diff
@@ -30,17 +27,26 @@ class VirtualizedSummaryView extends Component {
     section: sectionDataPropType.isRequired,
     scriptData: scriptDataPropType.isRequired,
     lessonOfInterest: PropTypes.number.isRequired,
-    getLevels: PropTypes.func,
+    levelsByLesson: PropTypes.object,
     onScroll: PropTypes.func,
     jumpToLessonDetails: PropTypes.func.isRequired
   };
 
-  state = {
-    fixedColumnCount: 1,
-    fixedRowCount: 1,
-    scrollToColumn: 0,
-    scrollToRow: 0
-  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      fixedColumnCount: 1,
+      fixedRowCount: 1,
+      scrollToColumn: 0,
+      scrollToRow: 0
+    };
+    this.summaryView = null;
```

Comment:
nice to see this. i'm using a ref in my current work and wasn't sure if we like to declare all properties even when they start as null. most of the examples i saw didn't do that so i didn't either, but i prefer your way so i will update my code!

Reply:
Hmm, yeah I think it's just personal preference. It's the old C programmer in me coming out.

Reply:
...or really any typesafe language... it's always made me uncomfortable that in js/python you can just add properties willy-nilly. 🙃